compiler/noirc_frontend/src/hir/resolution/traits.rs

```diff
@@ -207,16 +208,16 @@ fn collect_trait_impl_methods(
 
         if overrides.is_empty() {
             if let Some(default_impl) = &method.default_impl {
+                // copy 'where' clause from unresolved trait impl
+                let mut default_impl_clone = default_impl.clone();
+                default_impl_clone.def.where_clause = trait_impl.where_clause.clone();
```

Comment:
```suggestion
                default_impl_clone.def.where_clause.extend(trait_impl.where_clause.clone());
```
We should extend in case there are existing constraints on the default impl